backend/coins/eth/account.go

```diff
@@ -812,3 +814,133 @@ func (account *Account) VerifyAddress(addressID string) (bool, error) {
 func (account *Account) CanVerifyAddresses() (bool, bool, error) {
 	return account.Config().Keystore.CanVerifyAddress(account.Coin())
 }
+
+// SignMsg is used for personal_sign and eth_sign messages in BBApp via WalletConnect
+func (account *Account) SignMsg(
+	message string,
+) (string, error) {
+	bytesMessage, err := hex.DecodeString(strings.TrimPrefix(message, "0x"))
+	if err != nil {
+		return "", err
+	}
+	signedMessage, err := account.Config().Keystore.SignETHMessage(bytesMessage, account.signingConfiguration.AbsoluteKeypath())
+	if err != nil {
+		return "", err
+	}
+	return "0x" + hex.EncodeToString(signedMessage), nil
+}
+
+// SignTypedMsg signs an Ethereum EIP-612 typed message in BBApp via WalletConnect
+func (account *Account) SignTypedMsg(
+	chainId uint64,
+	data string,
+) (string, error) {
+	signedMessage, err := account.Config().Keystore.SignETHTypedMessage(chainId, []byte(data), account.signingConfiguration.AbsoluteKeypath())
+	if err != nil {
+		return "", err
+	}
+	return "0x" + hex.EncodeToString(signedMessage), nil
+}
+
+// TX proposal arguments received from Wallet Connect with Gas, GasPrice, Value and Nonce being optional
+type WalletConnectArgs struct {
+	From     string `json:"from"`
+	To       string `json:"to"`
+	Data     string `json:"data"`
+	Gas      string `json:"gas,omitempty"`
+	GasPrice string `json:"gasPrice,omitempty"`
+	Value    string `json:"value,omitempty"`
+	Nonce    string `json:"nonce,omitempty"`
+}
+
+// EthSignTypedMsg signs an Ethereum Tx received from WalletConnect
+func (account *Account) EthSignWalletConnectTx(
+	// send: whether transaction should be broadcast after signing
```

Comment:
maybe document the use case for send=false - i wonder how the caller can use a signature or txhash without knowing the transaction which is constructed inside this function but not returned and not broadcast.

Reply:
yeah, my bad, the WC docs weren't completely clear on this, but on second look it wants the raw transaction data which can later be submitted with sendRawTransaction
changed in the current fixup